include/mbedtls/check_config.h

```diff
@@ -76,13 +76,13 @@
 #endif
 
 #if defined(MBEDTLS_CMAC_C) && \
-    !defined(MBEDTLS_AES_C) && !defined(MBEDTLS_DES_C)
+    ( !defined(MBEDTLS_CIPHER_C ) || ( !defined(MBEDTLS_AES_C) && !defined(MBEDTLS_DES_C) ) )
 #error "MBEDTLS_CMAC_C defined, but not all prerequisites"
 #endif
 
 #if defined(MBEDTLS_NIST_KW_C) && \
     ( !defined(MBEDTLS_AES_C) || !defined(MBEDTLS_CIPHER_C) )
-#error "MBEDTLS_NIST_KW_C defined, but not all prerequisites"
+#error "MBEDTLS_NIST_KW_C defined, but not all prerequisites (missing CIPHER)"
```

Comment:
The "(missing CIPHER)" part looks like a leftover from debugging which should be removed from the final version (multiple occurrences).

Reply:
There are already error message that contains information about missing features:
`#error "MBEDTLS_PSA_CRYPTO_C defined, but not all prerequisites (missing RNG)"`.

I found this useful so added this for CIPHER also, but I can remove it.

Reply:
My main problem is, in that case the message isn't accurate: if `CIPHER_C` is enabled but `AES_C` isn't, we're still going to print "missing CIPHER" which is going to be confusing. I'm not against more explicit error messages, but they have to be accurate :)